src/routes/spaces/address-books.controller.spec.ts

```diff
@@ -198,6 +260,244 @@ describe('AddressBooksController', () => {
+
+      const firstItem = {
+        name: nameBuilder(),
+        address: getAddress(faker.finance.ethereumAddress()),
+        chainIds: faker.helpers.multiple(() => faker.string.numeric(), {
+          count: { min: 1, max: 5 },
+        }),
+      };
+
+      const secondItem = {
+        name: nameBuilder(),
+        address: getAddress(faker.finance.ethereumAddress()),
+        chainIds: faker.helpers.multiple(() => faker.string.numeric(), {
+          count: { min: 1, max: 5 },
+        }),
+      };
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({ items: [firstItem, secondItem] })
+        .expect(200);
+
+      const updatedFirstItem = {
+        ...firstItem,
+        name: nameBuilder(),
+      };
+
+      const { body: afterUpdateBody } = await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({ items: [updatedFirstItem, secondItem] })
+        .expect(200)
+        .expect(({ body }) =>
+          expect(body).toEqual({
+            spaceId: spaceId.toString(),
+            data: expect.arrayContaining([
+              {
+                ...updatedFirstItem,
+                createdBy: expect.any(String),
+                lastUpdatedBy: expect.any(String),
+              },
+              {
+                ...secondItem,
+                createdBy: expect.any(String),
+                lastUpdatedBy: expect.any(String),
+              },
+            ]),
+          }),
+        );
+
+      expect(afterUpdateBody.data.length).toBe(2);
+    });
+
+    it('should return a 404 if a space id does not exist', async () => {
+      const { accessToken } = await createSpace();
+      const nonExistingSpaceId = faker.number.int({
+        min: 69420,
+        max: DB_MAX_SAFE_INTEGER,
+      });
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${nonExistingSpaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({ items: [] })
+        .expect(404);
+    });
+
+    it('should return a 404 if the user does not exist', async () => {
+      const { spaceId } = await createSpace();
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const accessToken = jwtService.sign(authPayloadDto);
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .set('Cookie', [`access_token=${accessToken}`])
+        .send({ items: [] })
+        .expect(404);
+    });
+
+    it('should return a 403 if not authenticated', async () => {
+      const { spaceId } = await createSpace();
+
+      await request(app.getHttpServer())
+        .put(`/v1/spaces/${spaceId}/address-book`)
+        .send({ items: [] })
+        .expect(403);
+    });
+
+    it('should return a 404 if the member is not an admin', async () => {
```

Comment:
I understand that TypeORM throws because there is no member of `ADMIN` status, hence the 404 but this might make more sense as a 403. Let's check other tests as to how this is handled, and stick with 404 if it's a pattern.

Reply:
After fixing this I realized its causing other issues and requires quite a few changes i.e. injecting the membersRepository and making internal functions public so I will leave this out for now.